nilearn/decoding/decoder.py

```diff
@@ -330,6 +330,18 @@ def __init__(self, estimator='svc', mask=None, cv=10, param_grid=None,
         self.n_jobs = n_jobs
         self.verbose = verbose
 
+    def _check_estimator(self):
```

Comment:
A unit test for this function will likely reveal that the second condition never gets executed:
```python
         elif self.estimator in list(SUPPORTED_ESTIMATORS.keys()):
```
First, we need to check if the estimator is one of the supported estimators, and if not, if it is a string, and if not, then raise the error.
Instances such as these highlight the importance of writing unit tests for seemingly straightforward functions (I had almost decided a unit test for this function was not worth the hassle).

Also, keep these refactored functions below the calling function, so the code reader gets a higher-level overview of their use as they scroll down to see the details.